src/Entity/Dossier.php

```diff
@@ -136,6 +136,11 @@ public function getSousDossiers(): ArrayCollection|Collection|array
         return $this->sousDossiers;
     }
 
+    public function getVisibleSubFolders(bool $isOwner): ArrayCollection|Collection|array
+    {
+        return $isOwner ? $this->sousDossiers : $this->sousDossiers->filter(fn (Dossier $dossier) => !$dossier->isPrivate());
+    }
+
```

Comment:
Maybe we should rather pass User param instead, and check in this method if User is owner.
We could 
- avoid `$isOwner` var in controllers
- do `folder.visibleSubFolders(folder.beneficiary.user)` in template

Reply:
`folder.visibleSubFolders(folder.beneficiary.user)` will always be the owner 
But maybe you mean `folder.visibleSubFolders(app.user)`

Reply:
yes I did update my comment but maybe too late